Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -1722,8 +1722,12 @@ void WebPageProxy::maybeInitializeSandboxExtensionHandle(WebProcessProxy& proces
 #endif
     SandboxExtension::Handle sandboxExtensionHandle;
     if (!resourceDirectoryURL.isEmpty()) {
-        if (checkAssumedReadAccessToResourceURL && process.hasAssumedReadAccessToURL(resourceDirectoryURL))
+        if (checkAssumedReadAccessToResourceURL && process.hasAssumedReadAccessToURL(resourceDirectoryURL)) {
+#if PLATFORM(COCOA)
+            if (!sandbox_check(process.processID(), "file-read-data", static_cast<enum sandbox_filter_type>(SANDBOX_FILTER_PATH | SANDBOX_CHECK_NO_REPORT), resourceDirectoryURL.fileSystemPath().utf8().data()))
```

Comment:
I think it would be even better if there was a comment explaining why this check is needed.

Technically we should be using `fileSystemRepresentation` instead of `utf8`.

Reply:
I will update the PR with a comment and use `fileSystemRepresentation` instead.

Thanks for reviewing, all!